### PR TITLE
feat(tailwindcss-x): create tailwind plugin package

### PR DESCRIPTION
**Pull request by CachedaCodes**
[EX-5627](https://searchbroker.atlassian.net/browse/EX-5627)

## Motivation and context
Package for the Tailwind plugin with TS, PostCSS, and Vite. Added basic PostCSS plugins.

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [X] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that causes existing functionality to not work as expected)
- [ ] Change requires a documentation update

## What is the destination branch of this PR?

- [X] `Main`
- [ ] Other. Specify:

### CODE REVIEW

File: packages/tailwindcss-x/package.json [hunk 1]
```diff
@@ -0,0 +1,23 @@
+{
+  "name": "tailwindcss-x",
+  "private": true,
```
Comment (diegopf) [inline packages/tailwindcss-x/package.json:3]: I think we should add the description, author, license, homepage and similar fields like we do for other packages

File: packages/tailwindcss-x/package.json [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  "version": "0.1.0",
+  "scripts": {
+    "dev": "vite",
+    "build": "tsc && vite build",
+    "preview": "vite preview"
+  },
+  "devDependencies": {
+    "autoprefixer": "^10.4.4",
```
Comment (diegopf) [inline packages/tailwindcss-x/package.json:11]: I see that we are using the version `~9.7.4` can we keep both projects in sync? This way we will not have a package-lock.json here.

File: packages/tailwindcss-x/package.json [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    "cssnano": "^5.0.15",
+    "postcss": "^8.4.12",
+    "postcss-import": "^14.1.0",
+    "tailwindcss": "^3.0.23",
+    "typescript": "~4.6.2",
+    "vite": "^2.9.0"
```
Comment (diegopf) [inline packages/tailwindcss-x/package.json:17]: I think we should use always the ~ and not the ^ modifier for external dependencies versions

**Review by diegopf — CHANGES_REQUESTED** [review]
I see the readme.md file is missing, I think we should add one, shouldn't we?

File: .github/dependabot.yaml [hunk 4]
```diff
@@ -19,3 +19,4 @@ updates:
       day: "friday"
       time: "08:00"
       timezone: "Europe/Madrid"
+
```
Comment (diegopf) [inline .github/dependabot.yaml:22]: Remove empty line, please.

File: packages/tailwindcss-x/tsconfig.eslint.json [hunk 5]
```diff
@@ -0,0 +1,7 @@
+{
+  "extends": "./tsconfig.json",
+  "compilerOptions": {
+    "noEmit": true
```
Comment (diegopf) [inline packages/tailwindcss-x/tsconfig.eslint.json:4]: We already have this set on the parent tsconfig.json, so we should skip it here.

**Review by diegopf — CHANGES_REQUESTED** [review]

File: packages/tailwindcss-x/index.html [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+<html lang="en">
+  <head>
+    <meta charset="UTF-8" />
+    <link rel="icon" type="image/svg+xml" href="favicon.svg" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <link rel="stylesheet" href="/src/css/index.pcss" />
+    <title>tailwindcss-x</title>
+  </head>
+  <body>
+    <div id="app"></div>
```
Comment (tajespasarela) [inline packages/tailwindcss-x/index.html:11]: add an example to see tw working:

```html
    <button class="x-bg-amber-400 x-p-1 x-border-amber-500 x-border-2 x-border-solid x-rounded">
      Button
    </button>
```

File: packages/tailwindcss-x/tsconfig.eslint.json (hunk 5 above)
Comment (tajespasarela) [reply packages/tailwindcss-x/tsconfig.eslint.json:4]: actually, I think we are going to export types from this package so we have to remove that option from the parent.

File: packages/tailwindcss-x/index.html [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+  </body>
```
Comment (tajespasarela) [inline packages/tailwindcss-x/index.html:12]: Include the script that already imports the CSS:
```html
    <script type="module" src="src/main.ts"></script>
```

File: packages/tailwindcss-x/index.html [hunk 8]
```diff
@@ -0,0 +1,13 @@
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="UTF-8" />
+    <link rel="icon" type="image/svg+xml" href="favicon.svg" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <link rel="stylesheet" href="/src/css/index.pcss" />
```
Comment (tajespasarela) [inline packages/tailwindcss-x/index.html:7]: No need this as it is imported from the script

File: packages/tailwindcss-x/tsconfig.json [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+    "moduleResolution": "Node",
+    "strict": true,
+    "sourceMap": true,
+    "resolveJsonModule": true,
+    "esModuleInterop": true,
+    "noEmit": true,
+    "noUnusedLocals": true,
+    "noUnusedParameters": true,
+    "noImplicitReturns": true,
+    "skipLibCheck": true
```
Comment (tajespasarela) [inline packages/tailwindcss-x/tsconfig.json:16]: Include:
```js
    "importHelpers": true,
    "allowSyntheticDefaultImports": true,
    "declaration": true,
    "declarationMap": true,
```

and remove : 
```js
    "noEmit": true
```

**Review by tajespasarela — CHANGES_REQUESTED** [review]

**Review by tajespasarela — DISMISSED** [review]

File: packages/tailwindcss-x/tsconfig.json [hunk 10]
```diff
@@ -0,0 +1,23 @@
...
+    "moduleResolution": "Node",
+    "strict": true,
+    "sourceMap": true,
+    "resolveJsonModule": true,
+    "esModuleInterop": true,
+    "noUnusedLocals": true,
+    "noUnusedParameters": true,
+    "noImplicitReturns": true,
+    "skipLibCheck": true,
+    "importHelpers": true,
```
Comment (diegopf) [inline packages/tailwindcss-x/tsconfig.json:16]: Enabling importHelpers, now we need to install tslib as dependency. See https://www.typescriptlang.org/tsconfig#importHelpers

**Review by diegopf — CHANGES_REQUESTED** [review]

**Review by diegopf — APPROVED** [review]

**Review by tajespasarela — APPROVED** [review]

**Pull request merged**